### DISCUSSION TITLE
New design for CASL documentation

### DISCUSSION THREAD

**Discussion by enheit** [stalniy/casl#611] [Ideas]
Hi @stalniy,

First of all I want to say thank you for such a good lib and documentation that you (and team/community) support. 

Time moves, as do the requirements for the appearance of documentation (IMHO). I know that the main purpose of documentation is to describe the problem and guide newbies towards achieving their goals, but have you ever thought about redesigning your documentation site? If yes, what is the current status of this? If no, is it possible somehow to contribute?

I would like to try to investigate this topic and provide you an example of new version of the documentation (if you don't mind ofc)

What do you think?

Thank you, Roman

**Comment by avimehenwal** [comment]
Hello @enheit .
I am curious to know if you are suggesting only documentation style and presentation realted changes or data and content related changes ?

If they are content, explaination, examples or information on new-features realted then kindly put those topics/items in a list so that content writers can pick from there and contribute description to those topics.

If they are presentational changes only, you can suggest/recommend new template and wirefreames so that other contrinutors can get a feel of visual changes

**Comment by stalniy** [comment]
I think we need to understand the goal. Never thought about this. 

Current design is minimalistic but I’m open to suggestions

**Comment by stalniy** [comment]
would be awesome to replace handmade rollup config for docs with Vite.

**Comment by enheit** [comment]
> only documentation style and presentation realted changes or data and content related changes ?

The goal of this issue is to refresh the appearance of the site. I don't mean that the current visual design is bad, but it can definitely be improved.

Here is a list of inspirations that I had at the time of creating this issue

https://www.osohq.com/
https://mui.com/
https://tailwindcss.com/

> I think we need to understand the goal

The goal is simple. Improve the first impression of new users of the lib. 

I had a couple of ideas like putting the interactive examples on the main page to let people see the benefits right from the first touch.

I was able to find a small sketch of the main page of the CASL. It also includes a new logo (subject to discussion). I didn't try to make the whole concept at that moment

![photo_2022-08-28_16-42-34](https://user-images.githubusercontent.com/8645216/187076903-84d4eeba-c5ea-4c2b-a90e-e9219397a6aa.jpg)

**Reply by stalniy** [reply]
I like the idea of adding more interactive example, to let people play with smth like in Oso. (accepted)

Regarding new logo - makes sense but I need to sleep with that in mind for some time ))
